hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -257,46 +257,36 @@ public static void deleteInstantFile(FileSystem fs, String metaPath, HoodieInsta
 
   public void deleteEmptyInstantIfExists(HoodieInstant instant) {
     ValidationUtils.checkArgument(isEmpty(instant));
-    deleteInstantFileIfExists(instant);
+    deleteInstantIfExists(instant);
   }
 
   public void deleteCompactionRequested(HoodieInstant instant) {
     ValidationUtils.checkArgument(instant.isRequested());
     ValidationUtils.checkArgument(Objects.equals(instant.getAction(), HoodieTimeline.COMPACTION_ACTION));
-    deleteInstantFile(instant);
+    deleteInstantIfExists(instant);
   }
 
-  public void deleteInstantFileIfExists(HoodieInstant instant) {
-    LOG.info("Deleting instant " + instant);
-    Path inFlightCommitFilePath = getInstantFileNamePath(instant.getFileName());
-    try {
-      if (metaClient.getFs().exists(inFlightCommitFilePath)) {
-        boolean result = metaClient.getFs().delete(inFlightCommitFilePath, false);
```

Comment:
Seems the core change is renaming a method here: `deleteInstantFileIfExists` -> `deleteInstantIfExists` ? If that is true, i would suggest to keep as it is.

Reply:
no.
the major change is that, when need to delete active instant in archive operation, use HoodieActiveTimeline's `deleteInstantIfExists` instead of calling the naked api (fs.delete). After the changes, `HoodieActiveTimeline` becomes the standard route to operate active instants.

For this above, merge the origin method `deleteInstantFile` (used in current class and subclass) and `deleteInstantFileIfExists` (used outside) to one, and rename to `deleteInstantIfExists`.

Reply:
> calling the naked api (fs.delete).

`deleteInstantFile(HoodieInstant instant)` also takes a `HoodieInstant` param, so what's the difference here ?

Reply:
the old way to delete active instants when archive uses `deleteArchivedInstantFiles` -> `deleteFilesParallelize` in `HoodieTimelineArchiver`. and `deleteFilesParallelize` calls `fs.delete()` directly.